test/index.ts

```diff
@@ -1677,6 +1677,8 @@ describe('Datastore', () => {
             'metadata.otherProperty',
             'metadata.obj.*',
             'metadata.longStringArray[].*',
+            'undefinedData.*',
+            'undefinedArray[].*',
```

Comment:
We shouldn't modify the existing test. We should make a new test that tests for these specific values so we don't lose any existing test coverage.